base_portal_type/models/res_groups.py

```diff
@@ -24,51 +26,71 @@ def _update_user_groups_view(self):
         ][0]
         portal_group = self.env.ref("base.group_portal")
         internal_group = self.env.ref("base.group_user")
-        portal_groups = (
-            self.env["res.groups"]
-            .with_context(lang=None)
-            .search(
-                [
+
+        user_type_field_name = name_selection_groups(user_type_groups.ids)
+        non_public_attrs = str(
+            {
+                "readonly": [
                     (
-                        "category_id",
-                        "=",
-                        self.env.ref("base_portal_type.category_portal_type").id,
-                    ),
-                ]
-            )
-        )
-        user_type_field = name_selection_groups(user_type_groups.ids)
-        for node in arch.xpath("//group/field[@name='%s']" % user_type_field):
-            for group in portal_groups:
-                field_name = name_boolean_group(group.id)
-                for field_node in arch.xpath("//field[@name='%s']" % field_name):
-                    field_node.attrib["attrs"] = str(
-                        {
-                            "readonly": [
-                                (
-                                    user_type_field,
-                                    "not in",
-                                    (portal_group + internal_group).ids,
-                                )
-                            ],
-                        }
+                        user_type_field_name,
+                        "not in",
+                        [internal_group.id, portal_group.id],
                     )
-                node.addnext(
-                    etree.Element(
+                ],
+            }
+        )
+
+        portal_groups = self.env["res.groups"].search(
+            [
+                "|",
+                (
+                    "category_id",
+                    "=",
+                    self.env.ref("base_portal_type.category_portal_type").id,
+                ),
+                ("portal", "=", True),
+            ]
+        )
```

Comment:
```suggestion
        ).with_context(lang=None)
```

be sure that you use English for names